lib/src/main/java/co/nimblehq/recentapps/thumbnailhiding/RecentAppsThumbnailHidingActivity.kt

```diff
@@ -1,24 +1,55 @@
 package co.nimblehq.recentapps.thumbnailhiding
 
+import android.app.Activity
 import android.os.Build
 import android.os.Bundle
+import android.view.WindowManager
 import androidx.appcompat.app.AppCompatActivity
+import co.nimblehq.recentapps.thumbnailhiding.navbar.NavigationBarObserver.isGestureEnabled
 
-abstract class RecentAppsThumbnailHidingActivity : AppCompatActivity(), RecentAppsThumbnailHidingListener {
-
-    protected open val enableSecureFlagOnLowApiDevices: Boolean = false
+abstract class RecentAppsThumbnailHidingActivity : AppCompatActivity(),
+    RecentAppsThumbnailHidingListener {
 
     /**
      * HardwareKeyWatcher doesn't work on API 25 or lowers,
      * allow to use FLAG_SECURE instead to hide app thumbnail.
      */
-    val isSecureFlagEnabled: Boolean
+    open val enableSecureFlagOnLowApiDevices: Boolean = false
+
+    /**
+     * HardwareKeyWatcher doesn't work on some Xiaomi or Huawei devices with custom gesture navigation enabled,
+     * allow to use FLAG_SECURE instead to hide app thumbnail.
+     */
+    open val enableSecureFlagOnDevicesWithCustomGestureNavigation: Boolean = false
+
+    private val isSecureFlagOnLowApiDevicesEnabled: Boolean
         get() = enableSecureFlagOnLowApiDevices && Build.VERSION.SDK_INT < Build.VERSION_CODES.O
 
+    private val isSecureFlagOnDevicesWithCustomGestureNavigationEnabled: Boolean
```

Comment:
```suggestion
    private val isSecureFlagOnCustomGestureNavigationDevicesEnabled: Boolean
```
For consistency with `isSecureFlagOnLowApiDevicesEnabled`. 😁